tests/unit/multiclass/KNN_unittest.cc

```diff
@@ -226,3 +227,67 @@ TEST(KNN, classify_multiple_kdtree)
 
 
 }
+
+
+TEST(KNN, nearest_neighbours_brute)
```

Comment:
ok now that the nests are a bit cleaner, it is time to remove redundant code. Both tests are almost identical, so what you need to do is to use googletest to generate the code for both test cases for you. See eg this tutorial on how to do this:
http://sandordargo.com/blog/2019/04/24/parameterized-testing-with-gtest

It is also done in many shogun unit tests, so there are plenty of examples